packages/core/src/app/shipping/hooks/useShipping.ts

```diff
@@ -0,0 +1,130 @@
+import getShippableItemsCount from '../getShippableItemsCount';
+import getShippingMethodId from '../getShippingMethodId';
+import hasPromotionalItems from '../hasPromotionalItems';
+
+
+const deleteConsignmentsSelector = createSelector(
+    ({ checkoutService: { deleteConsignment } }: CheckoutContextProps) => deleteConsignment,
+    ({ checkoutState: { data } }: CheckoutContextProps) => data.getConsignments(),
+    (deleteConsignment, consignments) => async () => {
+        if (!consignments || !consignments.length) {
+            return;
+        }
+
+        const [{ data }] = await Promise.all(consignments.map(({ id }) => deleteConsignment(id)));
+
+        return data.getShippingAddress();
+    },
+);
+
+export const useShipping = () => {
+    const { checkoutState, checkoutService } = useCheckout();
+
+    const {
+        data: {
+            getCart,
+            getCheckout,
+            getConfig,
+            getCustomer,
+            getConsignments,
+            getShippingAddress,
+            getBillingAddress,
+            getShippingAddressFields,
+            getShippingCountries,
+        },
+        statuses: {
+            isShippingStepPending,
+            isSelectingShippingOption,
+            isLoadingShippingOptions,
+            isUpdatingConsignment,
+            isCreatingConsignments,
+            isCreatingCustomerAddress,
+            isLoadingShippingCountries,
+            isUpdatingBillingAddress,
+            isUpdatingCheckout,
+            isDeletingConsignment,
+            isLoadingCheckout,
+        },
+    } = checkoutState;
+
+    const checkout = getCheckout();
+    const config = getConfig();
+    const consignments = getConsignments() || [];
+    const customer = getCustomer();
+    const cart = getCart();
+
+    if (!checkout || !config || !customer || !cart) {
+        throw new Error('Unable to access checkout data');
+    }
+
+    const {
+        checkoutSettings: {
+            enableOrderComments,
+            hasMultiShippingEnabled,
+        },
+    } = config;
+
+    const methodId = getShippingMethodId(checkout, config);
+    const isLoading =
+        isLoadingShippingOptions() ||
+        isSelectingShippingOption() ||
+        isUpdatingConsignment() ||
+        isCreatingConsignments() ||
+        isUpdatingBillingAddress() ||
+        isUpdatingCheckout() ||
+        isCreatingCustomerAddress() ||
+        isDeletingConsignment() ||
+        isLoadingCheckout();
+
+    const shippableItemsCount = getShippableItemsCount(cart);
+    const shouldShowMultiShipping =
+        hasMultiShippingEnabled && !methodId && shippableItemsCount > 1;
+
+    const shippingAddress =
+        !shouldShowMultiShipping && consignments.length > 1 ? undefined : getShippingAddress();
+
+    const providerWithCustomCheckout = getProviderWithCustomCheckout(
+        config.checkoutSettings.providerWithCustomCheckout,
+    );
+
+    return {
```

Comment:
Things like cart, consignments, isGuest are not related to shipping, can we make `useShipping` for shipping only?